### PR TITLE
feat(frontend): syntax-highlight all tool results + expand language support

### PR DESCRIPTION
**Pull request by dimakis**
## Summary

- **All tool results now use `CodeBlock`** instead of plain `<pre>` — Bash output gets `bash` highlighting, Write/Edit results use the file's language, everything else gets consistent monospace styling with copy button
- **12 new languages** added to both `CodeBlock` registrations and `EXT_MAP`: clojure, dart, elixir, erlang, haskell, nginx, nix, objectivec, ocaml, powershell, scala, wasm
- Total language count: 34 → 46

## Test plan

- [ ] Expand a Read tool pill → file content highlighted (unchanged behaviour)
- [ ] Expand a Bash tool pill → output highlighted with bash colouring
- [ ] Expand a Write tool pill → result section uses CodeBlock styling
- [ ] Expand an Edit/Diff tool pill → result section uses CodeBlock styling
- [ ] Expand a Grep/Glob tool pill → plain monospace in CodeBlock (no language)
- [ ] Read/Write/Edit a `.nix`, `.ex`, `.hs`, `.dart`, `.scala` file → correct highlighting

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by dimakis** [comment]
## Centaur Review

Found **5** issue(s) (1 critical) (3 warning).


### `frontend/src/components/__tests__/ToolPill.test.tsx`
_Broken existing test for non-read tool rendering is the critical issue; the code change itself is sound but needs test updates for the new behavior and new EXT_MAP entries, plus an `nginx` sync gap between CodeBlock registrations and EXT_MAP._

- 🔴 **regressions** (L177): Existing test 'renders plain pre for non-read tools' (lines 177-192) will fail. It asserts `.tool-pill-pre` exists and `.code-block-highlight` is null — but the new code renders all tool results through CodeBlock, which uses `.code-block-highlight` and never creates `.tool-pill-pre`. The test must be updated to match the new unified rendering. `[fixable]`


### `frontend/src/components/CodeBlock.tsx`
_Broken existing test for non-read tool rendering is the critical issue; the code change itself is sound but needs test updates for the new behavior and new EXT_MAP entries, plus an `nginx` sync gap between CodeBlock registrations and EXT_MAP._

- 🔵 **style**: `nginx` is imported and registered in CodeBlock but has no extension mapping in `packages/protocol/src/language.ts` EXT_MAP. The comment at line 8 says 'Keep in sync with EXT_MAP'. Without an EXT_MAP entry (e.g., `conf: 'nginx'`), no file will ever resolve to this language via `languageFromPath`. Either add an EXT_MAP entry or remove the nginx registration. `[fixable]`


### `packages/protocol/__tests__/language.test.ts`
_Broken existing test for non-read tool rendering is the critical issue; the code change itself is sound but needs test updates for the new behavior and new EXT_MAP entries, plus an `nginx` sync gap between CodeBlock registrations and EXT_MAP._

- 🟡 **missing_tests**: The 12 new EXT_MAP entries (objectivec, dart, scala, haskell, ocaml, clojure, elixir, erlang, powershell, nix, wasm, and multi-extension variants like .cljs/.edn/.lhs/.mli/.hrl/.psm1/.psd1/.wast) have no test coverage. The existing test file covers 15 extensions but none of the newly added ones. At minimum, add one representative test per new language. `[fixable]`


### `frontend/src/components/ToolPill.tsx`
_Broken existing test for non-read tool rendering is the critical issue; the code change itself is sound but needs test updates for the new behavior and new EXT_MAP entries, plus an `nginx` sync gap between CodeBlock registrations and EXT_MAP._

- 🟡 **missing_tests**: No test covers the new behavior where a non-read tool with a language (e.g., a Bash/command tool with `language: 'bash'`) gets syntax highlighting via CodeBlock. The old test only covered the non-highlighted path. A test should verify that command-type tool results get bash highlighting and that Write-type results get their file-extension-derived highlighting. `[fixable]`
- 🟡 **regressions** (L93): Visual regression: non-read tool results previously rendered as a plain `<pre>` with a 'Result' `<span>` label (styled via `.tool-pill-label` + `.tool-pill-pre`). They now render inside a CodeBlock with a header bar containing label, copy button, and language badge. This changes the visual appearance — the header chrome (background, padding, copy button) will show even for simple one-line results like Grep output. Intentional UX change, but worth verifying it looks good for short/plain results.

File: packages/protocol/src/language.ts [hunk 1]
```diff
@@ -64,13 +83,21 @@ const EXT_MAP: Record<string, string> = {
...
+  psd1: 'powershell',
 
   // Markup / Docs
   md: 'markdown',
   mdx: 'markdown',
   tex: 'latex',
+
   // Infrastructure
   dockerfile: 'dockerfile',
+  conf: 'nginx',
```
Comment (dimakis) [inline packages/protocol/src/language.ts:97]: 🟡 **unsafe_assumptions**: Mapping `.conf` → `nginx` is too broad. Many non-nginx files use `.conf` (e.g. `httpd.conf` for Apache, `sysctl.conf`, `resolv.conf`, `logrotate.conf`, application configs). Applying nginx syntax highlighting to these will produce misleading results. Consider removing this mapping or using basename-level detection (e.g. only match files containing 'nginx' in the name). `[fixable]`

File: packages/protocol/src/language.ts [hunk 2]
```diff
@@ -56,6 +56,25 @@ const EXT_MAP: Record<string, string> = {
   kts: 'kotlin',
   swift: 'swift',
   cs: 'csharp',
+  m: 'objectivec',
```
Comment (dimakis) [inline packages/protocol/src/language.ts:59]: 🟡 **unsafe_assumptions**: Mapping `.m` → `objectivec` conflicts with MATLAB/Octave, which also uses `.m` files. In codebases that use MATLAB, this will produce incorrect highlighting. No great solution — just worth documenting the ambiguity as a code comment.

**Review by dimakis — COMMENTED** [review]
## Centaur Review

Found **3** issue(s) (2 warning).


### `packages/protocol/src/language.ts`
_Clean feature expansion — EXT_MAP and CodeBlock registrations are in sync, the ToolResult refactor handles all edge cases correctly, and tests cover the core behavior. Main concern is `.conf` → `nginx` being overly broad._

- 🟡 **unsafe_assumptions** (L97): Mapping `.conf` → `nginx` is too broad. Many non-nginx files use `.conf` (e.g. `httpd.conf` for Apache, `sysctl.conf`, `resolv.conf`, `logrotate.conf`, application configs). Applying nginx syntax highlighting to these will produce misleading results. Consider removing this mapping or using basename-level detection (e.g. only match files containing 'nginx' in the name). `[fixable]`
- 🟡 **unsafe_assumptions** (L59): Mapping `.m` → `objectivec` conflicts with MATLAB/Octave, which also uses `.m` files. In codebases that use MATLAB, this will produce incorrect highlighting. No great solution — just worth documenting the ambiguity as a code comment.


### `packages/protocol/__tests__/language.test.ts`
_Clean feature expansion — EXT_MAP and CodeBlock registrations are in sync, the ToolResult refactor handles all edge cases correctly, and tests cover the core behavior. Main concern is `.conf` → `nginx` being overly broad._

- 🔵 **missing_tests**: New extension variants added to EXT_MAP lack test coverage: `.mm` (objectivec), `.sbt` (scala), `.lhs` (haskell), `.mli` (ocaml), `.cljc`/`.edn` (clojure), `.exs` (elixir), `.hrl` (erlang), `.psm1`/`.psd1` (powershell), `.wast` (wasm). Each language has one extension tested but the secondary extensions are uncovered. `[fixable]`
